examples/poll_http_server/src/main.rs

```diff
@@ -0,0 +1,164 @@
+                    let (connection, address) = match server.accept(FDFLAGS_NONBLOCK) {
+                        Ok((connection, address)) => (connection, address),
+                        Err(ref e) if e.kind() == std::io::ErrorKind::WouldBlock => break,
+                        Err(e) => panic!("accept error: {}", e),
+                    };
+
+                    println!("Accepted connection from: {}", address);
+
+                    let token = unique_token.add();
+                    poll.register(&connection, token, Interest::Read);
+                    connections.insert(token, connection);
+                },
+                token => {
+                    let done = if let Some(connection) = connections.get_mut(&token) {
+                        let handler = match handlers.get_mut(&token) {
+                            Some(handler) => handler,
+                            None => {
+                                let handler = Handler::new();
+                                handlers.insert(token, handler);
+                                handlers.get_mut(&token).unwrap()
+                            }
+                        };
+                        handle_connection(&mut poll, connection, handler, &event)?
+                    } else {
+                        false
+                    };
+                    if done {
+                        if let Some(connection) = connections.remove(&token) {
+                            connection.shutdown(Shutdown::Both)?;
+                            poll.unregister(&connection);
+                            handlers.remove(&token);
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+fn handle_connection(
+    poll: &mut Poll,
+    connection: &mut TcpStream,
+    handler: &mut Handler,
+    event: &Event,
+) -> io::Result<bool> {
+    if event.is_readable() {
+        let mut connection_closed = false;
+        let mut received_data = vec![0; 4096];
+        let mut bytes_read = 0;
+        loop {
+            match connection.read(&mut received_data[bytes_read..]) {
+                Ok(0) => {
+                    connection_closed = true;
+                    break;
+                }
+                Ok(n) => {
+                    bytes_read += n;
+                    if bytes_read == received_data.len() {
+                        received_data.resize(received_data.len() + 1024, 0);
+                    }
+                }
+                Err(ref err) if would_block(err) => break,
+                Err(ref err) if interrupted(err) => continue,
+                Err(err) => return Err(err),
+            }
+        }
+
+        if bytes_read != 0 {
+            let received_data = &received_data[..bytes_read];
+            let mut bs: parsed::stream::ByteStream =
+                String::from_utf8(received_data.to_vec()).unwrap().into();
+            let req = parsed::http::parse_http_request(&mut bs).unwrap();
```

Comment:
You may receive an incomplete response. because L94 mean no data.

[TCP Half-Pack Problem](https://github.com/balloonwj/CppGuide/blob/master/articles/%E7%BD%91%E7%BB%9C%E7%BC%96%E7%A8%8B/TCP%E5%8D%8F%E8%AE%AE%E5%A6%82%E4%BD%95%E8%A7%A3%E5%86%B3%E7%B2%98%E5%8C%85%E3%80%81%E5%8D%8A%E5%8C%85%E9%97%AE%E9%A2%98.md)